packages/ui/ViewManager/TransitionGroup.js

```diff
@@ -293,10 +293,12 @@ class TransitionGroup extends React.Component {
 	}
 
 	completeTransition (key) {
-		delete this.currentlyTransitioningKeys[key];
+		if (key in this.currentlyTransitioningKeys) {
+			delete this.currentlyTransitioningKeys[key];
 
-		if (Object.keys(this.currentlyTransitioningKeys).length === 0) {
-			forwardOnTransition(null, this.props);
+			if (Object.keys(this.currentlyTransitioningKeys).length === 0) {
```

Comment:
Theoretically, this can still fire a spurious event if a panel is removed that was transitioning and a new transition will begin.  But, it's probably OK?